Tests/test_SffIO.py

```diff
@@ -368,181 +368,104 @@ def test_index2(self):
-        w = SffWriter(handle, xml=metadata)
-        w.write_file(iter(sff))
-        assert data == handle.getvalue()
+        with open(filename, "rb") as a_handle, open(filename, "rb") as b_handle:
+            metadata = ReadRocheXmlManifest(a_handle)
+            sff = list(SffIterator(b_handle))
+            b_handle = BytesIO()
+            w = SffWriter(b_handle, xml=metadata)
+            w.write_file(sff)  # list
+            data = b_handle.getvalue()
+            # And again with an iterator...
+            handle = BytesIO()
+            w = SffWriter(handle, xml=metadata)
+            w.write_file(iter(sff))
+            self.assertEqual(data, handle.getvalue())
         # Check 100% identical to the original:
-        filename = "Roche/E3MFGYR02_random_10_reads.sff"
         with open(filename, "rb") as handle:
             original = handle.read()
-            assert len(data) == len(original)
-            assert data == original
+            self.assertEqual(len(data), len(original))
+            self.assertEqual(data, original)
             del data
 
     def test_index(self):
         filename = "Roche/greek.sff"
-        with open(filename, "rb") as handle:
-            for record in SffIterator(handle):
-                print(record.id)
-        with open(filename, "rb") as handle:
-            index1 = sorted(_sff_read_roche_index(handle))
-        with open(filename, "rb") as handle:
-            index2 = sorted(_sff_do_slow_index(handle))
-        assert index1 == index2
+        # with open(filename, "rb") as handle:
+        #     for record in SffIterator(handle):
+        #         print(record.id)
+        with open(filename, "rb") as a_handle, open(filename, "rb") as b_handle:
+            index1 = sorted(_sff_read_roche_index(a_handle))
+            index2 = sorted(_sff_do_slow_index(b_handle))
+            self.assertEqual(index1, index2)
 
     def test_read_wrong(self):
         filename = "Roche/greek.sff"
-        try:
-            with open(filename, "rb") as handle:
-                # print(ReadRocheXmlManifest(handle))
-                r = ReadRocheXmlManifest(handle)
-            assert False, "Should fail!"
-        except ValueError:
-            pass
+        with open(filename, "rb") as handle:
+            # print(ReadRocheXmlManifest(handle))
+            self.assertRaises(ValueError, ReadRocheXmlManifest, handle)
 
         with open(filename, "rb") as handle:
             for record in SffIterator(handle):
                 pass
-            try:
+
+            def fileiter(handle):
                 for record in SffIterator(handle):
                     # print(record.id)
                     i = record.id
-                assert False, "Should have failed"
-            except ValueError as err:
-                pass
-                # print("Checking what happens on re-reading a handle:")
-                # print(err)
+            self.assertRaises(ValueError, fileiter, handle)
```

Comment:
Do we really need to define ``fileiter`` here? Does ``next(...)`` or even ``list(...)`` work by trying to start iteration through the file?